File: src/main/java/com/example/demo/service/TypeService.java
package com.example.demo.service;

import com.example.demo.model.dto.DictionaryData;
import com.example.demo.model.dto.DictionaryDataInput;

import java.util.List;

public interface TypeService {
    DictionaryData addType(DictionaryDataInput dictionaryDataInput);
    List<DictionaryData> findAllTypes();
    DictionaryData updateType(DictionaryData dictionaryData);
    DictionaryData findTypeById(Long id);
    void deleteType(Long id);
}

File: WEB/gun-mgmt-app/src/app/services/type.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, Type } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { DictionaryData } from '../types/DictionaryData';
import { DictionaryDataInput } from '../types/DictionaryDataInput';

@Injectable({
  providedIn: 'root'
})
export class TypeService {
  private apiEndpointUrl = `${environment.apiUrl}/types`;

  constructor(private http: HttpClient) { }

  public getAllTypes() :Observable<DictionaryData[]> {
    return this.http.get<DictionaryData[]>(`${this.apiEndpointUrl}`);
  }

  public getTypeById(id: number) :Observable<DictionaryData> {
    return this.http.get<DictionaryData>(`${this.apiEndpointUrl}/${id}`);
  }

  public addType(typeInput: DictionaryDataInput) :Observable<DictionaryData> {
    return this.http.post<DictionaryData>(`${this.apiEndpointUrl}/add`, typeInput);
  }

  public updateType(type: DictionaryData) :Observable<DictionaryData> {
    return this.http.put<DictionaryData>(`${this.apiEndpointUrl}/update`, type);
  }

  public deleteType(id: number) :Observable<void> {
    return this.http.delete<void>(`${this.apiEndpointUrl}/delete/${id}`);
  }
}
